salt/modules/cassandra_cql.py

```diff
@@ -0,0 +1,585 @@
+    :return:               Dictionary of permissions.
+    :rtype:                dict
+
+    .. code-block:: bash
+        salt 'minion1' cassandra.list_permissions
+
+        salt 'minion1' cassandra.list_permissions username=joe resource=test_keyspace permission=select
+
+        salt 'minion1' cassandra.list_permissions username=joe resource=test_table resource_type=table \
+        permission=select contact_points=minion1
+    '''
+    cql_template = "list %s on %s"
+
+    keyspace_cql = "%s %s" % (resource_type, resource) if resource else "all keyspaces"
+    permission_cql = "%s permission" % permission if permission else "all permissions"
+
+    query = cql_template % (permission_cql, keyspace_cql)
+    if username:
+        query = "%s of %s" % (query, username)
+    log.debug('Attempting to list permissions with query "%s"', query)
+    ret = _query(query, contact_points, port, cql_user, cql_pass)
+    return ret if ret else False
+
+def grant_permission(username, resource=None, resource_type='keyspace', permission=None, contact_points=None, port=None,
+                     cql_user=None, cql_pass=None):
+    '''
+    Grant permissions to a user.
+
+    :param username:       The name of the user to grant permissions to.
+    :type  username:       str
+    :param resource:       The resource (keyspace or table), if None, permissions for all resources are granted.
+    :type  resource:       str
+    :param resource_type:  The resource_type (keyspace or table), defaults to 'keyspace'.
+    :type  resource_type:  str
+    :param permission:     A permission name (e.g. select), if None, all permissions are granted.
+    :type  permission:     str
+    :param contact_points: The Cassandra cluster addresses, can either be a string or a list of IPs.
+    :type  contact_points: str | list[str]
+    :param cql_user:       The Cassandra user if authentication is turned on.
+    :type  cql_user:       str
+    :param cql_pass:       The Cassandra user password if authentication is turned on.
+    :type  cql_pass:       str
+    :param port:           The Cassandra cluster port, defaults to None.
+    :type  port:           int
+    :return:
+    :rtype:
+
+    .. code-block:: bash
+        salt 'minion1' cassandra.grant_permission
+
+        salt 'minion1' cassandra.grant_permission username=joe resource=test_keyspace permission=select
+
+        salt 'minion1' cassandra.grant_permission username=joe resource=test_table resource_type=table \
+        permission=select contact_points=minion1
+    '''
+    permission_cql = "grant %s" % permission if permission else "grant all permissions"
+    resource_cql = "on %s %s" % (resource_type, resource) if resource else "on all keyspaces"
+
+    query = "%s %s to %s" % (permission_cql, resource_cql, username)
+    log.debug('Attempting to grant permissions with query "%s"', query)
+    ret = _query(query, contact_points, port, cql_user, cql_pass)
+    return True if not ret else ret
```

Comment:
Please see the commend for the line 436. I would suggest not to do this way here as well. And the same would apply for the logging part as for commented line 543.